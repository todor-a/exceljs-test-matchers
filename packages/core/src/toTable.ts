import { Worksheet } from 'exceljs';
import { table } from 'table';

export const toTable = (worksheet?: Worksheet) => {
    const dataArray: Array<Array<string>> = [];

    worksheet?.eachRow((row) => {
        const rowData: Array<string> = [];
        row.eachCell((cell) => {
            rowData.push(cell.value as string);
        });
        dataArray.push(rowData);
    });

    return '\n' + table(dataArray, {
        border: {
            topBody: `─`,
            topJoin: `┬`,
            topLeft: `┌`,
            topRight: `┐`,

            bottomBody: `─`,
            bottomJoin: `┴`,
            bottomLeft: `└`,
            bottomRight: `┘`,

            bodyLeft: `│`,
            bodyRight: `│`,
            bodyJoin: `│`,

            joinBody: `─`,
            joinLeft: `├`,
            joinRight: `┤`,
            joinJoin: `┼`,
        },
    });
};
