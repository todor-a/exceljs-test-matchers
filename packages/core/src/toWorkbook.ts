import * as ExcelJS from 'exceljs';

export const toWorkbook = (data: Array<Array<string>>, sheet = 'data') => {
    const workbook = new ExcelJS.Workbook();

    const worksheet = workbook.addWorksheet(sheet);

    data.forEach((row_data, row_index) => {
        row_data.forEach((cell_data, col_index) => {
            worksheet.getCell(`${String.fromCharCode(65 + col_index)}${row_index + 1}`).value = cell_data;
        });
    });

    return workbook;
};
