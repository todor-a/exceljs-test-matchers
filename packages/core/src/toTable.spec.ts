import { expect, it } from 'vitest';
import { toTable } from './toTable';
import { toWorkbook } from './toWorkbook';

it('generates correct table', () => {
    const wb = toWorkbook(
        [
            ['a', 'b', 'c'],
            ['1', '2', '3'],
            ['4', '5', '6'],
        ],
        'demo',
    );
    expect(toTable(wb.getWorksheet('demo'))).toMatchSnapshot();
});
