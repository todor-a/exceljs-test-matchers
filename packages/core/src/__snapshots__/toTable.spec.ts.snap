// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generates correct table 1`] = `
"
┌───┬───┬───┐
│ a │ b │ c │
├───┼───┼───┤
│ 1 │ 2 │ 3 │
├───┼───┼───┤
│ 4 │ 5 │ 6 │
└───┴───┴───┘
"
`;
